### PR TITLE
An initial test setup

### PR DESCRIPTION
**Pull request by theAnuragMishra**
This pr adds a pseudo-random events generator perl script for the purposes of dev and testing.

Usage:
from the repo root, run `./scripts/generate-seed-data.sh`. 
For options see `./scripts/generate-seed-data.sh --help`.

### CODE REVIEW

**Comment by Thunder-Blaze** [comment]
I think we could use `Cpanel::JSON::XS` while keeping `JSON::PP` as fallback in case someone wants to generate high amount of seed data to test

https://stackoverflow.com/questions/14591444/which-perl-module-would-you-recommend-for-json-manipulation

**Comment by theAnuragMishra** [comment]
Is complicating really worth it... I didn't feel it was slow generating 100k events.

**Comment by Thunder-Blaze** [comment]
> Is complicating really worth it... I didn't feel it was slow generating 100k events.

I don't think it's that complicated, and as I said it'll be better in general for the users if they want to generate larger dataset (1-10M) not 100K

**Comment by hoytech** [comment]
I agree, there's no need to optimise for performance here.

Since this will only be used in the test suite (right?) can we please keep it all contained in the `test/` directory, for clarity?

File: test/filterFuzzTest.pl [hunk 1]
```diff
@@ -269,17 +193,17 @@ sub testMonitor {
...
 
 if ($cmd eq 'scan') {
-    while (1) {
+    for (1..100) {
         my $fg = genRandomFilterGroup();
         testScan($fg);
     }
 } elsif ($cmd eq 'scan-limit') {
-    while (1) {
+    for (1.100) {
```
Comment (hoytech) [inline test/filterFuzzTest.pl]: Missing a `.`?

File: test/filterFuzzTest.pl [hunk 2]
```diff
@@ -5,100 +5,24 @@
...
-bitcoin
-nos
-nostr
-nostrnovember
-gitlog
-introductions
-jb55
-damus
-chat
-nosuchtopic
```
Comment (hoytech) [inline test/filterFuzzTest.pl:100]: `nosuchtopic` is there to test for topics that for sure won't have matches, maybe we can continue to push something similar on too?

File: test/generate-seed-data.pl [hunk 3]
```diff
@@ -0,0 +1,630 @@
...
+    my ($current_idx) = @_;
+    my $count = scalar @users;
+    return 0 if $count <= 1;
+    my $idx = rng_int($count - 1);
+    $idx++ if $idx >= $current_idx;
+    return $idx;
+}
+
+sub next_ts {
+    $clock += 1 + rng_int(4);
```
Comment (hoytech) [inline test/generate-seed-data.pl]: There are actually a lot of edge cases where events have the same timestamp. So maybe just get rid of the `1 +` here. Then there will be lots of events with duplicate timstamps.

File: test/filterFuzzTest.pl (hunk 2 above)
Comment (theAnuragMishra) [reply test/filterFuzzTest.pl:100]: added!

File: test/filterFuzzTest.pl (hunk 1 above)
Comment (theAnuragMishra) [reply test/filterFuzzTest.pl]: fixed!
